src/lib/ast.rs

```diff
@@ -349,13 +363,28 @@ impl NodeId {
         self.index
     }
 
+    /// Get the height of this node.
+    ///
+    /// The height of a leaf node is 1, the height of a branch node is 1 +
+    /// the height of its tallest child node.
+    pub fn height<T: Clone>(&self, arena: &Arena<T>) -> u32 {
+        if arena[*self].is_leaf() {
+            return 1;
+        }
+        let mut heights: Vec<u32> = self.children(arena)
+            .map(|child| child.height(arena))
+            .collect();
```

Comment:
You might find it more natural to use the `let heights = self.children(...).map(...).collect::<Vec<u32>>()` syntax.